packages/components/app/styles/components/pagination.scss

```diff
@@ -0,0 +1,177 @@
+  --token-pagination-indicator-height: 2px;
+  --token-pagination-indicator-width: 19px;
+}
+
+// * Pagination Parent (wrapper) component
+// ----------------------------------------
+.hds-pagination {
+  display: grid;
+  grid-template-areas: "left center right";
+  grid-template-rows: auto;
+  grid-template-columns: 1fr auto 1fr;
+  align-items: center;
+  margin: 0 auto;
+}
+
+// * Sub-component: Info
+// ---------------------------------------
+.hds-pagination-info {
+  grid-area: left;
+  justify-self: flex-start;
+  white-space: nowrap;
+}
+
+// * Sub-component: Nav
+// ----------------------------------------
+.hds-pagination-nav {
+  display: flex;
+  grid-area: center;
+}
+
+.hds-pagination-nav__page-list {
+  display: flex;
+  margin: 0;
+  padding: 0;
+}
+
+.hds-pagination-nav__page-item { list-style-type: none; }
+
+// nav controls: button-arrow, button-number, ellipsis:
+.hds-pagination-nav__control,
+.hds-pagination-nav__ellipsis {
+  display: flex;
+  align-items: center;
+  padding: 0 var(--token-pagination-control-padding-horizontal);
+}
+
+.hds-pagination-nav__control {
+  height: var(--token-pagination-control-height);
+  text-decoration: none;
+  background: transparent;
+  border: none;
+
+  @include hds-focus-ring-with-pseudo-element(
+    $top: var(--token-pagination-control-focus-inset),
+    $right: var(--token-pagination-control-focus-inset),
+    $bottom: var(--token-pagination-control-focus-inset),
+    $left: var(--token-pagination-control-focus-inset),
+  );
+
+  &:hover,
+  &.mock-hover {
+    color: var(--token-color-foreground-action-hover);
+  }
+
+  &:active,
+  &.mock-active {
+    color: var(--token-color-foreground-action-active);
+  }
+
+  &:disabled,
+  &.mock-disabled {
+    color: var(--token-color-foreground-disabled);
+    cursor: not-allowed;
+  }
+}
+
+// Make buttons with "Previous" & "Next" labels the same min-width
+.hds-pagination-nav__button-arrow--labeled {
+  min-width: 103px;
+}
+
+.hds-pagination-nav__button-arrow--direction-prev {
+  flex-direction: row;
+  justify-content: flex-start;
+
+  .hds-pagination-nav__page-label {
+    margin-left: 6px;
+  }
+}
+
+.hds-pagination-nav__button-arrow--direction-next {
+  flex-direction: row-reverse;
+  justify-content: flex-end;
+
+  .hds-pagination-nav__page-label {
+    margin-right: 6px;
+  }
+}
+
+.hds-pagination-nav__ellipsis {
+  color: var(--token-color-foreground-faint);
+}
+
+.hds-pagination-nav__button-number--is-selected {
+  position: relative;
+  color: var(--token-color-foreground-action);
+
+  &:hover {
+    color: var(--token-color-foreground-action-hover);
+  }
+
+  &:active {
+    color: var(--token-color-foreground-action-active);
+  }
+
+  .hds-pagination-nav__control {
+    &,
+    &:hover,
+    &:active {
+      color: inherit;
+    }
+  }
+
+  // indicator underline
+  &::before {
+    position: absolute;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    width: var(--token-pagination-indicator-width);
```

Comment:
this fixed width is not resilient. what if the text is 4 chars? (eg. page 1234)? in that case the width would be be correct (unless by design is fixed).
 if instead is proportional to the actual width of the container, then the `left/right` properties are more apt for this purpose

Reply:
Above browser, below Figma
<img width="721" alt="screenshot_2209" src="https://user-images.githubusercontent.com/686239/208080004-2424b805-374e-4e60-8c83-f283e81eb8b0.png">